### PR TITLE
TLS enhancements

### PR DESCRIPTION
**Pull request by dougm**
- Support custom TLS certificate verification

- Add govc about.cert command

- Add govc host.cert commands

### CODE REVIEW

**Comment by fdawg4l** [comment]
LGTM

File: govc/about/cert.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	info := new(object.HostCertificateInfo).FromCertificate(r.peer)
+
+	if r.cmd.thumbprint {
+		u := r.cmd.URLWithoutPassword()
+		_, err := fmt.Fprintf(w, "%x %s\n", sha1.Sum([]byte(u.Host)), info.ThumbprintSHA1)
+		return err
+	}
+
+	if r.err == nil {
+		info.Status = string(types.HostCertificateManagerCertificateInfoCertificateStatusGood)
```
Comment (caglar10ur) [inline govc/about/cert.go]: could do 

```
info.Status = string(types.HostCertificateManagerCertificateInfoCertificateStatusGood)
if r.err ! nil {
    info.Status = fmt.Sprintf("ERROR %s", r.err)
}
```

File: govc/flags/client.go [hunk 2]
```diff
@@ -331,6 +367,53 @@ func (flag *ClientFlag) loadClient() (*vim25.Client, error) {
...
+
+	if _, ok := verr.(x509.UnknownAuthorityError); !ok {
+		return verr
+	}
+
+	f, err := os.Open(flag.tlsKnownHosts)
+	if err != nil {
+		return err
+	}
+
```
Comment (caglar10ur) [inline govc/flags/client.go]: extra line

File: govc/flags/client.go [hunk 3]
```diff
@@ -331,6 +367,53 @@ func (flag *ClientFlag) loadClient() (*vim25.Client, error) {
...
+			continue
+		}
+
+		if w.ThumbprintSHA1 == e[1] {
+			return nil
+		}
+
+		return fmt.Errorf("Host %q fingerprint does not match %s:%d", name, flag.tlsKnownHosts, line)
+	}
+
```
Comment (caglar10ur) [inline govc/flags/client.go]: do we need to check scanner error here?

```
if err := scanner.Err(); err != nil {
    bla
}
```

File: object/host_certificate_info.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	info.subjectName = &cert.Subject
+	info.issuerName = &cert.Issuer
+
+	info.Issuer = info.fromName(info.issuerName)
+	info.NotBefore = &cert.NotBefore
+	info.NotAfter = &cert.NotAfter
+	info.Subject = info.fromName(info.subjectName)
+	info.Status = string(types.HostCertificateManagerCertificateInfoCertificateStatusUnknown)
+
+	{
```
Comment (caglar10ur) [inline object/host_certificate_info.go]: scoped cause you wanted to use sum and hex as variable names? 💯  :)

**Comment by caglar10ur** [comment]
LGTM

**Comment by dougm** [comment]
@caglar10ur thanks, happy to address those

**Pull request merged**